src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -270,7 +270,7 @@ internal async Task<IEnumerable<IRpcWorkerChannel>> GetInitializedWorkerChannels
                 }
             }
             IEnumerable<IRpcWorkerChannel> workerChannels = webhostChannels == null ? _jobHostLanguageWorkerChannelManager.GetChannels() : webhostChannels.Union(_jobHostLanguageWorkerChannelManager.GetChannels());
-            IEnumerable<IRpcWorkerChannel> initializedWorkers = workerChannels.Where(ch => ch.State == RpcWorkerChannelState.Initialized);
+            IEnumerable<IRpcWorkerChannel> initializedWorkers = workerChannels.Where(ch => ch.State.HasFlag(RpcWorkerChannelState.Initialized) && ch.State.HasFlag(RpcWorkerChannelState.InvocationBuffersInitialized));
```

Comment:
Instead of exposing "State" directly and expecting else to know which combinations to look for, can we add a method to RpcWorkerChannel like `IsReadyForInvocation()`. Since there are many combinations of flags but only a few valid ones, it would be nice to only expose the information we need to to other classes.

Reply:
good idea! will take a look.